### ISSUE TITLE
error:  (linebreak): invalid node with type whatsit and subtype 19 found in discretionary

### ISSUE THREAD

**Issue by jtojnar** [latex3/luaotfload#191]
When trying to build the following document with LuaLaTeX on Ubuntu (using TeXLive from latest Nixpkgs), it fails on my computer with the error in the title:

```latex
\documentclass{article}
\usepackage{hyperref}
\usepackage{polyglossia}

\setmainfont{Gentium Plus}

\begin{document}
See also \href{https://en.wiktionary.org/}{Wiktio}.
\end{document}
```

Weirdly, it works if I change the link text to just `Wikti`, remove the link or use a different font.

I was **unable** to reproduce it on Overleaf so perhaps it has something to do with fonts installed on my computer? I tried clearing `~/.texlive2021/` but it did not help.

Any idea what does the message mean or how to interpret it?

Full log:

```ShellSession
$ lualatex  pisma.tex
This is LuaHBTeX, Version 1.13.0 (Web2C 2021/NixOS.org) 
 restricted system commands enabled.
(./pisma.tex
LaTeX2e <2020-10-01> patch level 4
 L3 programming layer <2021-02-18>
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/latex/base/article.cls
Document Class: article 2020/04/10 v1.4m Standard LaTeX document class

(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/latex/base/size10.clo))
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/latex/hyperref/hyperref.sty
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/generic/ltxcmds/ltxcmds.sty)
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/generic/iftex/iftex.sty)
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/generic/pdftexcmds/pdftexcmds.sty
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/generic/infwarerr/infwarerr.sty))
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/latex/graphics/keyval.sty)
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/generic/kvsetkeys/kvsetkeys.sty)
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/generic/kvdefinekeys/kvdefinekeys.sty)
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/generic/pdfescape/pdfescape.sty)
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/latex/hycolor/hycolor.sty)
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/latex/letltxmacro/letltxmacro.sty)
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/latex/auxhook/auxhook.sty)
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/latex/kvoptions/kvoptions.sty)
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/latex/hyperref/pd1enc.def)
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/latex/hyperref/hyperref-langpatches.def)
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/generic/intcalc/intcalc.sty)
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/generic/etexcmds/etexcmds.sty)
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/latex/hyperref/puenc.def)
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/latex/url/url.sty)
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/generic/bitset/bitset.sty
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/generic/bigintcalc/bigintcalc.sty))
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/latex/base/atbegshi-ltx.sty))
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/latex/hyperref/hluatex.def
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/generic/stringenc/stringenc.sty)
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/latex/base/atveryend-ltx.sty)
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/latex/rerunfilecheck/rerunfilecheck.sty
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/generic/uniquecounter/uniquecounter.sty)))
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/latex/polyglossia/polyglossia.sty
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/latex/etoolbox/etoolbox.sty)
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/latex/makecmds/makecmds.sty)
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/latex/xkeyval/xkeyval.sty
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/generic/xkeyval/xkeyval.tex
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/generic/xkeyval/xkvutils.tex)))
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/latex/fontspec/fontspec.sty
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/latex/l3packages/xparse/xparse.sty
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/latex/l3kernel/expl3.sty
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/latex/l3backend/l3backend-luatex.def))
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/latex/l3packages/xparse/xparse-2020-10-01.sty
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/latex/l3packages/xparse/xparse-generic.tex)))
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/latex/fontspec/fontspec-luatex.sty
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/latex/base/fontenc.sty)
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/latex/fontspec/fontspec.cfg)))
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/latex/l3packages/l3keys2e/l3keys2e.sty)
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/luatex/luatexbase/luatexbase.sty
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/luatex/ctablestack/ctablestack.sty))
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/latex/polyglossia/gloss-latex.ldf))
No file pisma.aux.

(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/latex/base/ts1cmr.fd)
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/latex/hyperref/nameref.sty
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/latex/refcount/refcount.sty)
(/nix/store/xygf45hbpwbrwhixhrpi4kdp6dyikd5g-texlive-combined-2021/share/texmf/
tex/generic/gettitlestring/gettitlestring.sty))
! error:  (linebreak): invalid node with type whatsit and subtype 19 found in d
iscretionary
!  ==> Fatal error occurred, no output PDF file produced!
```

----

Initially [reported](https://tex.stackexchange.com/questions/603442/error-linebreak-invalid-node-with-type-whatsit-and-subtype-19-found-in-discr) on TeX StackExchange where I was asked to open an issue here.

**Comment by jtojnar** [comment]
These are the aux/log files produced:

[pisma.log](https://github.com/latex3/luaotfload/files/6751388/pisma.log)
[pisma.aux](https://github.com/latex3/luaotfload/files/6751389/pisma.aux.txt)

And these are the font files in question:
[fonts.zip](https://github.com/latex3/luaotfload/files/6751395/fonts.zip)

**Comment by jtojnar** [comment]
Does not seem to happen when I upgrade to Gentium 6 system-wide.

**Comment by zauguin** [comment]
Thanks for the report. This is fixed in b7672cce61dc7cd4d7f8227d155c3c994e63165d. (So it will be fixed with the next release.)

**Comment by zauguin** [comment]
Fixed in the latest release.
